railties/lib/rails/generators/rails/plugin/plugin_generator.rb

```diff
@@ -432,7 +432,7 @@ def rails_app_path
       end
 
       def inside_application?
-        rails_app_path && app_path =~ /^#{rails_app_path}/
+        rails_app_path && destination_root =~ /^#{rails_app_path}/
```

Comment:
If `rails_app_path` is nil, `start_with?` always returns true(e.g. `"a".start_with?(nil.to_s) #=> true`). Therefore, I think that it is necessary to check nil.